canine/adapters/base.py

```diff
@@ -69,22 +107,34 @@ def parse_inputs(self, inputs: typing.Dict[str, typing.Union[typing.Any, typing.
         Returns a job input specification useable for Localization
         Also sets self.spec to the same dictionary
         """
+
+        #Pin fixed arrays
+        inputs = {
+            key: _FixedArray(val) if key in self.arrays else val
```

Comment:
2D arrays provided as inputs should be implicitly resolved to FixedArrays here. Having to explicitly specify which inputs should be arrays via a separate `arrays` parameter is unfriendly.

Reply:
Alternatively, this logic could happen upstream in the orchestrator. Not sure about the best place to put it.

Reply:
What about this idea, which would remove the need to declare arrays:
* If a 1D array is provided as an input, it's considered a list of values for different jobs, as normal.
* If a 2D (or more) array is provided as an input, it is considered a list of arrays for different jobs. The list is flattened to 2 dimensions and one sub-list is provided to each job
* If a list of any dimensions is provided as an input which is marked as common, the array is flattened, but it is provided to all jobs.

The only problem with that implementation is that it requires the adapter to have knowledge of the localizer overrides, but in the short term, it wouldn't be hard just to provide the list of inputs marked as common. In the long term, we have discussed the idea of merging adapters and localizers and I'm still keen on the idea

Reply:
>If a list of any dimensions is provided as an input which is marked as common, the array is flattened, but it is provided to all jobs.

This should also implicitly be the case if a 2D array is provided as input, but it has length 1, e.g. `[[foo, bar, baz]]`. I think as-is this will automagically be handled as a common input?